Dockerfile

```diff
@@ -0,0 +1,20 @@
+FROM openjdk:8
+
+# add a non-privileged user for running the application
+RUN groupadd --gid 10001 app && \
+    useradd -g app --uid 10001 --shell /usr/sbin/nologin --create-home --home-dir /app app
+
+WORKDIR /app
+
+# Install python
+RUN apt-get update && \
+    apt-get -y --no-install-recommends install python2.7 python-pip python-setuptools
+
+# Finally copy in the app's source file
+COPY . /app
+
+ENV PYTHONPATH $PYTHONPATH:/app/usage_report:/app/tests
+
+RUN pip install -r requirements.txt
```

Comment:
For better layer caching, I'd recommend copying in just `requirements.txt` before running this line, and then run the broader `COPY . /app`. That should ensure that installing python packages only gets repeated if `requirements.txt` changes.